### ISSUE TITLE
The build fails randomly when in a CI/CD pipeline

### ISSUE THREAD

**Issue by sneko** [getsentry/sentry-webpack-plugin#372]
### Environment

How do you use Sentry?
SaaS

Which SDK and version?
Node v12

### Steps to Reproduce

Hi,

I'm running a pipeline that was working 100% of the time before adding the `@sentry/webpack-plugin`. Now it's failing randomly always at the same step.

The difficulty is there is no log at all.

Since my Jenkins is deploying on Kubernetes with some logic around `jnlp` containers and `node` ones I cannot affirm the issue comes from your plugin itself. But not having logs is quite annoying.

### Expected Result

That it works :)

### Actual Result

When it works I get something like:

```
yarn run build --mode dev
yarn run v1.22.4
$ cross-env NODE_ENV=production vue-cli-service build --mode dev

-  Building for dev...
Starting type checking and linting service...
Using 1 worker with 2048MB memory limit
Browserslist: caniuse-lite is outdated. Please run:
npx browserslist@latest --update-db
> Found 394 release files
> Analyzing 394 sources
> Rewriting sources
> Adding source map references
> Bundled 394 files for upload
> Uploaded release files to Sentry
> File upload complete (processing pending on server)
> Organization: XXXXXX
> Project: XXXXX
> Release: XXXXXXx
> Dist: None

Source Map Upload Report
  Scripts
    ~/precache-manifest.2070c4c8647a2d5fd50c7bc120045416.js
    ~/service-worker.js
  Minified Scripts
    ~/js/app.b5d98875.js (sourcemap at app.b5d98875.js.map)
    ~/js/chunk-01c203a5.efb601f9.js (sourcemap at chunk-01c203a5.efb601f9.js.map)
    ~/js/chunk-01e28590.474b1a56.js (sourcemap at chunk-01e28590.474b1a56.js.map)
```

But when it randomly fails I only get:

```
yarn run build --mode dev
yarn run v1.22.4
$ cross-env NODE_ENV=production vue-cli-service build --mode dev

-  Building for dev...
Starting type checking and linting service...
Using 1 worker with 2048MB memory limit
Browserslist: caniuse-lite is outdated. Please run:
npx browserslist@latest --update-db
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

I tried to make sure of `silent: false` and `debug: true` but it does not brings much more information:
```
yarn run build --mode dev
yarn run v1.22.4
$ cross-env NODE_ENV=production vue-cli-service build --mode dev

-  Building for dev...
[Sentry Webpack Plugin] DEBUG: Injecting release code
[Sentry Webpack Plugin] DEBUG: Loaders:
 [
  [32m'/YYYYYYYYYYYYYYYYYYYYY/node_modules/vue-loader/lib/loaders/pitcher.js'[39m,
  [32m'/YYYYYYYYYYYYYYYYYYYYY/node_modules/url-loader/dist/cjs.js'[39m,
  [32m'<unknown loader>'[39m,
  [32m'/YYYYYYYYYYYYYYYYYYYYY/node_modules/url-loader/dist/cjs.js'[39m,
  [32m'/YYYYYYYYYYYYYYYYYYYYY/node_modules/url-loader/dist/cjs.js'[39m,
  [32m'<unknown loader>'[39m,
  [32m'<unknown loader>'[39m,
  [32m'<unknown loader>'[39m,
  [32m'<unknown loader>'[39m,
  [32m'<unknown loader>'[39m,
  [32m'<unknown loader>'[39m,
  [32m'<unknown loader>'[39m,
  [32m'/YYYYYYYYYYYYYYYYYYYYY/node_modules/@vue/cli-plugin-babel/node_modules/thread-loader/dist/cjs.js'[39m,
  [32m'/YYYYYYYYYYYYYYYYYYYYY/node_modules/cache-loader/dist/cjs.js'[39m,
  [32m'/YYYYYYYYYYYYYYYYYYYYY/node_modules/cache-loader/dist/cjs.js'[39m,
  [32m'graphql-tag/loader'[39m,
  [32m'webpack-strip-block'[39m,
  [32m'/YYYYYYYYYYYYYYYYYYYYY/node_modules/cache-loader/dist/cjs.js'[39m,
  [32m'/YYYYYYYYYYYYYYYYYYYYY/node_modules/url-loader/dist/cjs.js'[39m,
  [32m'<unknown loader>'[39m,
  [32m'/YYYYYYYYYYYYYYYYYYYYY/node_modules/url-loader/dist/cjs.js'[39m,
  [32m'/YYYYYYYYYYYYYYYYYYYYY/node_modules/url-loader/dist/cjs.js'[39m,
  [32m'<unknown loader>'[39m,
  [32m'<unknown loader>'[39m,
  [32m'<unknown loader>'[39m,
  [32m'<unknown loader>'[39m,
  [32m'<unknown loader>'[39m,
  [32m'<unknown loader>'[39m,
  [32m'<unknown loader>'[39m,
  [32m'/YYYYYYYYYYYYYYYYYYYYY/node_modules/@vue/cli-plugin-babel/node_modules/thread-loader/dist/cjs.js'[39m,
  [32m'/YYYYYYYYYYYYYYYYYYYYY/node_modules/cache-loader/dist/cjs.js'[39m,
  [32m'/YYYYYYYYYYYYYYYYYYYYY/node_modules/cache-loader/dist/cjs.js'[39m,
  [32m'graphql-tag/loader'[39m,
  [32m'webpack-strip-block'[39m,
  [32m'/YYYYYYYYYYYYYYYYYYYYY/node_modules/@sentry/webpack-plugin/src/sentry.loader.js'[39m
]
[Sentry Webpack Plugin] DEBUG: Added loaders:
 [
  [32m'/YYYYYYYYYYYYYYYYYYYYY/node_modules/@sentry/webpack-plugin/src/sentry.loader.js'[39m
]
[Sentry Webpack Plugin] DEBUG: Removed loaders:
 []
[Sentry Webpack Plugin] DEBUG: Entry:
 {
  app: [
    [32m'/YYYYYYYYYYYYYYYYYYYYY/node_modules/@sentry/webpack-plugin/src/sentry-webpack.module.js'[39m,
    [32m'./src/main.ts'[39m
  ]
}
[Sentry Webpack Plugin] DEBUG: Added entry:
 [
  {
    app: [
      [32m'/YYYYYYYYYYYYYYYYYYYYY/node_modules/@sentry/webpack-plugin/src/sentry-webpack.module.js'[39m,
      [32m'./src/main.ts'[39m
    ]
  }
]
[Sentry Webpack Plugin] DEBUG: Removed entry:
 [ { app: [ [32m'./src/main.ts'[39m ] } ]
Starting type checking and linting service...
Using 1 worker with 2048MB memory limit
Browserslist: caniuse-lite is outdated. Please run:
npx browserslist@latest --update-db
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

Note that by removing Sentry plugin it works.

So I decided to move from the plugin version `v1.18.8` to `1.18.9` to test Sentry CLI `v2.x` instead of `v1.x`... but still getting the same error.

I tried to hardcode debug logs into your `index.js` and I see all logs passing across `apply(compiler)`, that's after that it fails. I guess probably because in the meantime `apply` is giving some future work through `attachAfterCodeGenerationHook` and `attachAfterEmitHook`.

My logs in the callback `cb` of `attachAfterEmitHook` are never written. So I guess something is blocking before that but I cannot figure it out for now.

I'm wondering if it could be the communication with the CLI that times out or something else. But I would like to understand what's going wrong... looking for logs haha

I also tried to specify `errorHandler` to be sure it was shown:
```
          errorHandler: (err, invokeErr, compilation) => {
            console.log(err.message);
            compilation.warnings.push('Sentry CLI Plugin: ' + err.message);
          },
```

If you have any thought it would be great, after spending hours on this I'm quite stuck :)

Thank you,

cc @kamilogorek 

PS: note that I have no `yarn-error.log` that is created unfortunately

**Comment by kamilogorek** [comment]
I wonder if it's caused by the `sentry-cli` itself, or something in between.

To verify that, you can set `SENTRY_LOG_LEVEL=debug` in your CI, and the logs should produce something more useful.

If they don't, it means that `sentry-cli` binary was not even called at that point.

**Comment by kamilogorek** [comment]
I'm out for a week now, so won't be able to debug this right away. Will try to take a look once I'm back.

**Comment by sneko** [comment]
Hi @kamilogorek :)

Any thought since then? I'm still stuck with this, it was not something temporary unfortunately :(

**Comment by sneko** [comment]
Hi,

I saw this after setting Sentry, but yesterday I switched from Node v12 to Node v16 and it was very frequent. After searching a while I found it's due to memory limits. Unfortunately I don't know why I never got any proper logs from Node (since Node didn't crash abruptly)...

If you want the whole context I wrote it down: https://stackoverflow.com/a/72542194/3608410

Sorry for this open issue!

@kamilogorek @daryawood
